tests/test_basic_functionality.py

```diff
@@ -343,6 +344,36 @@ def test_dictionary_style_access() -> None:
     assert lct["test_key"] == "test_value"
 
 
+def test_dunder_key_attr_style_access() -> None:
+    lct = LayeredConfigTree()
+    lct.update({"__dunder_key__": "val"})
+    assert lct["__dunder_key__"] == "val"
+
+    with pytest.raises(
+        RuntimeError,
+        match=re.escape(
+            "Cannot get an attribute starting and ending with '__' via attribute "
+            "access (i.e. dot notation). Use dictionary access instead "
+            "(i.e. bracket notation)."
+        ),
+    ):
+        lct.__dunder_key__
+
+    with pytest.raises(
+        RuntimeError,
+        match=re.escape(
+            "Cannot set an attribute starting and ending with '__' via attribute "
+            "access (i.e. dot notation). Use dicationary access instead "
+            "(i.e. bracket notation)."
+        ),
+    ):
+        lct.__dunder_key__ = "val2"
+    assert lct["__dunder_key__"] == "val"
```

Comment:
Maybe good to add a step where you update the value with bracket syntax (allowed)

Reply:
You can't, you need to go through `.update()`

Reply:
I'm confused -- why can't you `lct["__dunder_key__"] = "val3"`? That's using the bracket syntax and I don't see anywhere we block that.

Reply:
Well I guess it's a bit more complicated than I implied.

You can't directly call `lct["existing_key"] = "some_new_thing"` because [you can't overwrite an existing layer](https://github.com/ihmeuw/layered_config_tree/blob/main/src/layered_config_tree/main.py#L173-L179). You'd need to go through the `lct.update()` method.

Reply:
oh, I guess I could instantiate a lct with multiple layers

Reply:
I honestly find the restrictions here so confusing... would love to write them down in order to fully get it